packages/common/http/src/params.ts

```diff
@@ -73,7 +73,7 @@ export class HttpUrlEncodingCodec implements HttpParameterCodec {
 function paramParser(rawParams: string, codec: HttpParameterCodec): Map<string, string[]> {
   const map = new Map<string, string[]>();
   if (rawParams.length > 0) {
-    const params: string[] = rawParams.split('&');
+    const params: string[] = rawParams.replace('?', '').split('&');
```

Comment:
I think to be safe, we should only strip the `?` from the start of the string yes?
So

```suggestion
    const params: string[] = rawParams.replace(/^?/, '').split('&');
```

Reply:
You're totally right, thanks for the feedback will change it right away.

Reply:
Should the regex not be `/[\\^?]/`

Reply:
Sorry it should be `/^\?/` (I forgot to escape the question mark.) This is using `^` here to mark the start of the string.
In `/[\\^?]/` this is saying match any character that is one of `\`, '^` or `?`.

Reply:
So just to be sure, add an extra test that checks that a question mark later in the param list is not replaced.

Reply:
Thank you for your explanation 🙏 pushing it while typing 😉:wq